src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorTest.php

```diff
@@ -149,11 +127,11 @@ public function testValidateIfChildWithValidConstraint()
         $this->assertNoViolation();
     }
 
-    public function testDontValidateIfParentWithoutCascadeValidation()
+    public function testDontValidateIfParentWithoutValidConstraint()
```

Comment:
@peterrehm I guess the naming of the tests is confusing. We have two different cases:

* the `constraints` option: Constraints set in this option are *always* validated (given their groups match).
* the `Valid` constraint: This constraint (either set in the validation metadata - e.g. as annotation - or in the `constraints` option) results in the constraints defined ine the metadata of the *data* being validated.

Example:

~~~php
// data_class = Customer

$form->add('address', FormType::class);
$form->get('address')->add('street', TextType::class, [
    'constraints' => new NotBlank(),
]);

class Customer
{
    private $address;
}

class Address
{
    /**
     * @Length(min=3)
     */
    private $street;
}
~~~

In this example, by default only the `NotBlank` constraint is validated. The constraints defined in the `Address` constraints, however, are not validated:

 * There is no `Valid` constraint in the `constraints` option of the `address` field in the form.
 * There is no `Valid` constraint on the `Customer::$address` property.

If you add the `Valid` constraint in either of these places, the `Length` constraint is validated as well.

We could clarify the tests for example by renaming `testValidateConstraintsEvenIfNoValidConstraint` to `testValidateConstraintsOptionEvenIfParentWithoutValidConstraint` etc.